Integrate platform build instructions into documentation

TcT2k opened the pull request:
Integrate the various build instructions currently located in various documents in *docs/<platform>* into the doxygen documentation.

I've already done this for MSW und OSX/iOS but just wanted to collect some feedback before including the other platforms.

ToDo:
- [x] MSW
- [x] OS X
- [x] iOS
- [x] QT
- [x] GTK
- [x] Motif
- [x] X11

vadz commented on docs/doxygen/platforms/msw_install.md:
```diff
@@ -46,22 +37,22 @@ actually required, this makes using the library more convenient and
-    will get compilation errors if any new symbols are added to
-    setup0.h file in the repository.
+didn't obtain them from a release file, you also need to copy
+*include/wx/msw/setup0.h* to *include/wx/msw/setup.h* and to remember
```
Minor, but I think we should use backticks instead of asterisks around the file names.

vadz commented on docs/doxygen/platforms/msw_install.md:
```diff
@@ -381,54 +375,54 @@ depending on the compiler used.
 The full list of the build settings follows:
 
-BUILD=release
+* BUILD=release <br/>
```
Maybe use sub-subsections instead of an itemized list and avoid the need for the (ugly) `<br/>`?

vadz reviewed (commented):
Globally this looks good to me, thanks a lot for improving this!

However I think we still need to keep `docs/$platform/readme.txt` files because they appear in many mailing list and blog posts etc. At the very least, they should contain "Please refer to the <link> section of the manual", but I wonder if we couldn't keep them there, instead of moving them under `docs/doxygen` and still include them from Doxygen? This would be slightly illogical (some of Doxygen docs will be outside of its directory) but I think it can be pretty convenient to have these files in a simple to find place under docs and so keeping them there would be preferable. Renaming them from `.txt` to `.md` shouldn't be a problem, of course.

Thanks again!

TcT2k committed:
Integrate MSW, OS X and iOS build instructions into doxygen

Make build instructions available in doxygen but keep the old
folder structure.

TcT2k committed:
Integrate GTK documentation into doxygen

TcT2k committed:
Integrate QT documentation into doxygen

TcT2k committed:
Integrate X11 and Motif install instructions into doxygen

TcT2k commented:
@vadz I've left the files in their original location now. The platform paths are specified in the Doxyfile explicitly as simply including `../` would result in a recursion because the `doxygen` folder is on the same level. The readme files are untouched as they are not referenced from the documentation.

TcT2k committed:
Add information on how to use pre-built MSW binaries to doxygen

The text is collected from docs/contributing/release.md and the
blog post explaining usage of binaries.

vadz commented:
Thanks again for improving these docs, this is very important and useful!

I'm merging this, but just a small problem: my Vim highlights all occurrences of underscores (`_`) in the new `.md` files in red ("markdownError"). I don't know if it's really a problem or not, but I wonder if we could use dashes (`-`) instead of underscores in the links? As for the underscores in file/directory names, I can just put single backquotes around them.
